test/integration/build_to_buildruns_test.go

```diff
@@ -0,0 +1,298 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package integration_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/test"
+)
+
+const (
+	BUILD    = "build-"
+	BUILDRUN = "buildrun-"
+	STRATEGY = "strategy-"
+)
+
+var _ = Describe("Integration tests Build and BuildRuns", func() {
+
+	var (
+		cbsObject      *v1alpha1.ClusterBuildStrategy
+		buildObject    *v1alpha1.Build
+		buildRunObject *v1alpha1.BuildRun
+		buildSample    []byte
+		buildRunSample []byte
+	)
+
+	// Load the ClusterBuildStrategies before each test case
+	BeforeEach(func() {
+		cbsObject, err = tb.Catalog.LoadCBSWithName(STRATEGY+tb.Namespace, []byte(test.ClusterBuildStrategySingleStep))
+		Expect(err).To(BeNil())
+
+		err = tb.CreateClusterBuildStrategy(cbsObject)
+		Expect(err).To(BeNil())
+
+	})
+	// Delete the ClusterBuildStrategies after each test case
+	AfterEach(func() {
+
+		_, err = tb.GetBuild(buildObject.Name)
+		if err == nil {
+			Expect(tb.DeleteBuild(buildObject.Name)).To(BeNil())
+		}
+
+		err := tb.DeleteClusterBuildStrategy(cbsObject.Name)
+		Expect(err).To(BeNil())
+	})
+
+	// Override the Builds and BuildRuns CRDs instances to use
+	// before an It() statement is executed
+	JustBeforeEach(func() {
+		if buildSample != nil {
+			buildObject, err = tb.Catalog.LoadBuildWithNameAndStrategy(BUILD+tb.Namespace, STRATEGY+tb.Namespace, buildSample)
+			Expect(err).To(BeNil())
+		}
+
+		if buildRunSample != nil {
+			buildRunObject, err = tb.Catalog.LoadBRWithNameAndRef(BUILDRUN+tb.Namespace, BUILD+tb.Namespace, buildRunSample)
+			Expect(err).To(BeNil())
+		}
+	})
+
+	Context("when a build with a short timeout is defined", func() {
+
+		BeforeEach(func() {
+			buildSample = []byte(test.BuildCBSWithShortTimeOut)
+			buildRunSample = []byte(test.MinimalBuildRun)
+		})
+
+		It("should fail the builRun with a Reason", func() {
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
```

Comment:
I do not really have a strong opinion on this, but I like to use `HaveOccurred()` to `Expect` where I know the return value is an error, mostly for readability.
```suggestion
			Expect(tb.CreateBuild(buildObject)).ToNot(HaveOccurred())
```

Reply:
Now that I read the suggestion when it is rendered here, it actually reads more confusing with `ToNot` `HaveOccurred`.

Reply:
Would be interested in other feedback on this. I am fine with `BeNil` here.

Reply:
I think I was doing this based on the typical go code block that looks like:
```
if err != nil {

}
```
Im fine keeping it the way it is.